Lib/unittest/test/testmock/testasync.py

```diff
@@ -512,19 +560,30 @@ def test_mock_async_for(self):
             return accumulator
 
         expected = ["FOO", "BAR", "BAZ"]
-        with self.subTest("iterate through default value"):
-            mock_instance = MagicMock(self.WithAsyncIterator())
-            self.assertEqual([], asyncio.run(iterate(mock_instance)))
+        def test_default(mock_type):
+            mock_instance = mock_type(self.WithAsyncIterator())
+            self.assertEqual(asyncio.run(iterate(mock_instance)), [])
+
 
-        with self.subTest("iterate through set return_value"):
-            mock_instance = MagicMock(self.WithAsyncIterator())
+        def test_set_return_value(mock_type):
+            mock_instance = mock_type(self.WithAsyncIterator())
             mock_instance.__aiter__.return_value = expected[:]
-            self.assertEqual(expected, asyncio.run(iterate(mock_instance)))
+            self.assertEqual(asyncio.run(iterate(mock_instance)), expected)
 
-        with self.subTest("iterate through set return_value iterator"):
-            mock_instance = MagicMock(self.WithAsyncIterator())
+        def test_set_return_value_iter(mock_type):
+            mock_instance = mock_type(self.WithAsyncIterator())
             mock_instance.__aiter__.return_value = iter(expected[:])
-            self.assertEqual(expected, asyncio.run(iterate(mock_instance)))
+            self.assertEqual(asyncio.run(iterate(mock_instance)), expected)
+
+        for mock_type in [AsyncMock, MagicMock]:
```

Comment:
I understand this construction but I would prefer something like that, but we don't use pytest. (maybe a new feature for the unittest lib?)
```python
# @pytest.mark.parametrize('mock_type', (AsyncMock, MagicMock))
@unittest.parametrize('mock_type', (AsyncMock, MagicMock))
def test_default(self, mock_type):
    pass
```

Reply:
Not a bad idea! I think there are a number of things we could borrow from pytest. Parameterization would make this kind of construction much nicer.